qemu/tests/smt_test.py

```diff
@@ -34,6 +34,10 @@ def get_guest_threads():
         else:
             cmd = params["get_cores_cmd"]
             output = session.cmd_output_safe(cmd)
+            if not output:
+                output = session.cmd_output_safe(cmd)
+                if not output:
+                    test.error("Failed to get cores inside guest")
```

Comment:
How about warp a function to retry command, no only for `get_cores_cmd`
And to avoid the index error issue, i suggest:
```
match = re.search(r"=(\d+)", output)
if match:
    cores = int(match.group(1))
else:
    test.error(****)
```

Reply:
Seems only the first command returns empty, I'm not sure why and if we need retry for every command. What do you think?

Reply:
We can use retry times of the function to control which command need run multiple times, How do you think ?

Reply:
@yanan-fu I added a function with a parameter 'retry', pls help review. Test passed on my host. Thanks.